repos/fdbt-reference-data-service/src/uploaders/txc_uploader/txc_processor.py

```diff
@@ -358,6 +436,72 @@ def all_journey_pattern_sections_are_not_empty(data: dict, service: dict) -> boo
         and all_journey_pattern_sections_are_not_empty(data, service)
     )
 
+def get_vehicle_journeys(data_dict):
+    if (
+        "VehicleJourneys" in data_dict["TransXChange"]
+        and data_dict["TransXChange"]["VehicleJourneys"] is not None
+    ):
+        vehicle_journeys = make_list(
+            data_dict["TransXChange"]["VehicleJourneys"]["VehicleJourney"]
+        )
+
+        return vehicle_journeys
+
+def collect_vehicle_journey(vehicle):
+    vehicle_journey_info = {
+        "vehicle_journey_code": vehicle["VehicleJourneyCode"]
+        if "VehicleJourneyCode" in vehicle
+        else None,
+        "service_ref": vehicle["ServiceRef"] if "ServiceRef" in vehicle else None,
+        "line_ref": vehicle["LineRef"] if "LineRef" in vehicle else None,
+        "journey_pattern_ref": vehicle["JourneyPatternRef"]
+        if "JourneyPatternRef" in vehicle
+        else None,
+    }
+
+    return vehicle_journey_info
+
+def format_vehicle_journeys(vehicle_journeys: list, line_id: str):
+    vehicle_journey_refs = [
+        journey["VehicleJourneyRef"]
+        for journey in vehicle_journeys
+        if journey["LineRef"] == line_id
+        and "JourneyPatternRef" not in journey
+        and "VehicleJourneyRef" in journey
+        and "LineRef" in journey
+    ]
+    vehicle_journeys_for_line = [
+        journey
+        for journey in vehicle_journeys
+        if "JourneyPatternRef" in journey
+        and (
+            ("LineRef" in journey and journey["LineRef"] == line_id)
+            or (
+                "VehicleJourneyCode" in journey
+                and journey["VehicleJourneyCode"] in vehicle_journey_refs
+            )
+        )
+    ]
+
+    vehicle_journeys_data = []
```

Comment:
`journey_pattern_count` won't be needed here since that's used for tracks data so this line onwards could be simplified to:

```python
return [
    collect_vehicle_journey(vehicle_journey)
    for vehicle_journey in vehicle_journeys_for_line
]
```

Reply:
done